lib/fastlane/plugin/appcenter/actions/appcenter_fetch_version_number.rb

```diff
@@ -0,0 +1,99 @@
+require 'json'
+require 'net/http'
+require 'fastlane_core/ui/ui'
+
+module Fastlane
+  UI = FastlaneCore::UI unless Fastlane.const_defined?("UI")
+
+  module Actions
+    class AppcenterFetchVersionNumberAction < Action
+      def self.description
+        "Fetches the latest version number of an app from App Center"
+      end
+
+      def self.authors
+        ["jspargo", "ShopKeep"]
+      end
+
+      def self.run(params)
+        api_token = params[:api_token]
+        app_name = params[:app_name]
+        owner_name = params[:owner_name]
+
+        if owner_name.nil?
+          owner_name = get_owner_name(api_token, app_name)
+        end
+
+        releases = Helper::AppcenterHelper.fetch_releases(
+          api_token: api_token,
+          owner_name: owner_name,
+          app_name: app_name
+        )
+
+        UI.abort_with_message!("No versions found for '#{app_name}' owned by #{owner_name}") unless releases
+        sorted_release = releases.sort_by { |release| release['id'] }.reverse!
+        latest_build = sorted_release.first
+
+        if latest_build.nil?
```

Comment:
Defensive coding can't hurt, but is this possible? You're checking there are at least some releases above already. Or can that be an empty array?

Reply:
Yeh, I actually [added a spec](https://github.com/microsoft/fastlane-plugin-appcenter/pull/186/files#diff-f3feb2c2c170312dcd04d3e48f0f2611R75-R87) for this case too. Because you may have created an app but not yet uploaded any releases, you get an empty array back from the API, so I wanted to handle that as gracefully as possible. Let me know of you think this is over-engineered though?

Reply:
Since it's covering something that can happen, keep it like this.